### ISSUE TITLE
`-Wimplicit-fallthrough` suggests to use a macro from a random third-party library instead of `[[fallthrough]]`

### ISSUE THREAD

**Issue by HolyBlackCat** [llvm/llvm-project#71397]
Here's the code:
```cpp
#define FMT_FALLTHROUGH [[fallthrough]]

void a();

void b(int x)
{
    switch (x)
    {
      case 1:
        a();
      case 2:
        a();
    }
}
```
When compiled with Clang 17 (or trunk), it says:
```none
<source>:11:7: warning: unannotated fall-through between switch labels [-Wimplicit-fallthrough]
   11 |       case 2:
      |       ^
<source>:11:7: note: insert 'FMT_FALLTHROUGH;' to silence this warning
   11 |       case 2:
      |       ^
      |       FMT_FALLTHROUGH; 
<source>:11:7: note: insert 'break;' to avoid fall-through
   11 |       case 2:
      |       ^
      |       break; 
```
Notice how it suggests using `FMT_FALLTHROUGH` instead of `[[fallthrough]]`, presumably because it finds a macro that expands to this attribute.

Now this is clever in theory, but in my case, the macro comes from a third-party library (libfmt in this case), and I wouldn't want to use their internal macro in my code.

I suggest that this warning should suggest always suggest `[[fallthrough]]`, never a macro.

**Comment by dwblaikie** [comment]
Existing behavior was implemented for codebases that use a macro version of `[[fallthrough]]` for portability reasons - so that'll need to be balanced with the inverse problem you've described. Not sure what the right answer is.

**Comment by HolyBlackCat** [comment]
@dwblaikie One idea that comes to mind is to make a command line switch to select the preferred spelling, and to default to `[[fallthrough]]` if not specified.

**Comment by dwblaikie** [comment]
Yeah, maybe something like that would be feasible. Not sure it'll be top of anyone's list to implement any of this, though, unfortunately. If you're feeling up to it, could try prototyping a patch and sending a pull request.
